polaris-react/src/components/Text/Text.tsx

```diff
@@ -48,7 +48,7 @@ export interface TextProps {
   color?: Color;
   /** Adjust weight of text */
   fontWeight?: FontWeight;
-  /** Truncate text overflow with ellipsis */
+  /** Truncate text overflow with ellipsis when `as` prop is set to a block element */
```

Comment:
Taking a look at this implementation it makes sense to me. I wonder if truncate should just apply `display: block`
https://github.com/seek-oss/braid-design-system/blob/master/lib/components/private/Truncate/Truncate.tsx

Reply:
Yes the intent behind https://github.com/Shopify/polaris/pull/6797/files#diff-8ee4c2ce9fa2a15a8575923c63afff900480c13e0a001dc1007929d1fedf67e7L75 was that truncate would automatically also apply block. Maybe removing it was conflating that with alignment? We don't want alignment to apply block, but truncate is ok?

Reply:
My only question here before we change it back would be what are the cons of allowing Text to use truncate with an inline element as opposed to block?

Reply:
> what are the cons of allowing Text to use truncate with an inline element
It could create confusion when truncate is set, but nothing happens

> I feel like truncate and align should apply block as they don't work without it.
That was our original intent before the discussion to decouple them in #6797 😂

Shall I close this PR then?

Reply:
Oh I see. I am so confusing sometimes. I think the intent in 6797 is right. If a developer wants to use `align` or `truncate` it should apply block. HOWEVER it shouldn't apply block by default. Only when those properties are used. The issue before was that `alignment = 'inherit'` was setting a default value and causing everything to be `block` by default. I don't think that should be the case. We can close this PR but should...

1. Set no default value to alignment, or only set display block if alignment !== `inherit`
2. Keep the block style for when alignment or truncate is set
3. This should allow `span` without alignment or truncate to be inline by default

Hope this makes sense @aveline